source/fundamentals/crud/compound-operations.txt

```diff
@@ -0,0 +1,313 @@
+===================
+Compound Operations
+===================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide you will learn how to perform **compound operations** with the
+MongoDB Java driver. 
+
+Compound operations consist of a read and write operation performed as one
+**atomic operation**. An atomic operation is an operation which either completes
+entirely, or does not complete at all. Atomic operations cannot partially complete.
+
+Atomic operations can help you avoid **race conditions** in your code. A
+race condition occurs when your code's behavior is dependent on the order of
+uncontrollable events.
```

Comment:
```suggestion
 events.
```
Uncontrollable seems wordy. Removing it still delivers the same message.

Reply:
Thanks for the note. I believe it is incorrect to say that this edit delivers the same meaning. All programs (and all happenings within a computer) are dependent on events. A person must start a program (an event), a program must move information within a machine (an event), etc... "Uncontrollable" is the main word in the definition of race condition here.